src/libPMacc/include/cuSTL/algorithm/kernel/run-time/Foreach.hpp

```diff
@@ -130,10 +130,10 @@ math::Size_t<DIM3> getBestCudaBlockDim(const math::Size_t<dim> gridDim)
         if(this->_blockDim == math::Size_t<DIM3>::create(0))                                                        \
             this->_blockDim = getBestCudaBlockDim(p_zone.size);                                                     \
                                                                                                                     \
-        assert(this->_blockDim.productOfComponents() <= cudaSpecs::maxNumThreadsPerBlock);                          \
-        assert(this->_blockDim.x() <= cudaSpecs::MaxNumThreadsPerBlockDim::x::value);                               \
-        assert(this->_blockDim.y() <= cudaSpecs::MaxNumThreadsPerBlockDim::y::value);                               \
-        assert(this->_blockDim.z() <= cudaSpecs::MaxNumThreadsPerBlockDim::z::value);                               \
+        PMACC_VERIFY(this->_blockDim.productOfComponents() <= cudaSpecs::maxNumThreadsPerBlock);                    \
+        PMACC_VERIFY(this->_blockDim.x() <= cudaSpecs::MaxNumThreadsPerBlockDim::x::value);                         \
+        PMACC_VERIFY(this->_blockDim.y() <= cudaSpecs::MaxNumThreadsPerBlockDim::y::value);                         \
+        PMACC_VERIFY(this->_blockDim.z() <= cudaSpecs::MaxNumThreadsPerBlockDim::z::value);                         \
```

Comment:
please add spaces `PMACC_VERIFY( this->_blockDim.z() <= cudaSpecs::MaxNumThreadsPerBlockDim::z::value ); `